controllers/component_dependency_update_controller.go

```diff
@@ -351,16 +387,31 @@ func (r *ComponentDependencyUpdateReconciler) handleCompletedBuild(ctx context.C
 			componentsToUpdate = append(componentsToUpdate, comp)
 		}
 	}
-	var nudgeErr error
 
+	updatedOutputImage := updatedComponent.Spec.ContainerImage
+	imageRepositoryHost := strings.Split(updatedOutputImage, "/")[0]
+
+	imageRepositoryUsername, imageRepositoryPassword, err := r.getImageRepositoryCredentials(ctx, pipelineRun.Namespace, updatedOutputImage)
```

Comment:
What if image repository is public?

Reply:
even public repository will have push secret